File: A1/A1-015/A1-015.cpp
#include<iostream>
using namespace std;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);

    // Input
    string first_name, last_name, age;
    cin >> first_name >> last_name >> age;
    // Construct password
    string password = "";
    if(first_name.size() > 5 && last_name.size() > 5)
    {
        password += first_name[0];
        password += first_name[1];
        password += last_name.back();
        password += age.back();
    }
    else
    {
        password += first_name[0];
        password += age;
        password += last_name.back();
    }
    // Output
    cout << password;
    return 0;
}

File: A1/A1-015/A1-015.py
# --------------------------------------------------
# File Name : A1-015.py
# Problem   : Password
# Author    : Worralop Srichainont
# Date      : 2026-02-08
# --------------------------------------------------

# Input name and age
first_name = input().strip()
last_name = input().strip()
age = input().strip()

# Output password
if len(first_name) > 5 and len(last_name) > 5:
    print(f"{first_name[:2]}{last_name[-1]}{age[-1]}")
else:
    print(f"{first_name[0]}{age}{last_name[-1]}")
